### PR TITLE
Update to .NET 6 Preview 7

### LINKED ISSUE #25148: Removed pinned dependency to Microsoft.Extensions.Caching.Memory.
https://github.com/dotnet/efcore/pull/25092 added a pinned dependency to Microsoft.Extensions.Caching.Memory, which we'll need to update before shipping

@AndriySvyryd

### PR DESCRIPTION
**Pull request by bricelam**
Resolves #25148

### CODE REVIEW

**Review by smitpatel — DISMISSED** [review]

**Comment by bricelam** [comment]
Ugg. OData failures. (again)

**Comment by bricelam** [comment]
@ajcvickers Looks like there are binary breaking changes in Logging: 😕

> MissingMethodException : Method not found: 'Microsoft.Extensions.Logging.LoggerMessage.Define(LogLevel, EventId, string, bool)'.

File: test/EFCore.OData.FunctionalTests/Properties/TestAssemblyConditions.cs [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
 using Microsoft.EntityFrameworkCore.TestUtilities;
+using Microsoft.EntityFrameworkCore.TestUtilities.Xunit;
 
 // Skip the entire assembly if not on Windows and no external SQL Server is configured
 [assembly: SqlServerConfiguredCondition]
+
+[assembly: LoggingIsBrokenCondition]
+
+[AttributeUsage(AttributeTargets.Assembly)]
+public sealed class LoggingIsBrokenConditionAttribute : Attribute, ITestCondition
```
Comment (smitpatel) [inline test/EFCore.OData.FunctionalTests/Properties/TestAssemblyConditions.cs:15]: 👀

Comment (bricelam) [reply test/EFCore.OData.FunctionalTests/Properties/TestAssemblyConditions.cs:15]: Better to add `Skip = "..."` in all the test attributes?

Comment (bricelam) [reply test/EFCore.OData.FunctionalTests/Properties/TestAssemblyConditions.cs:15]: I tried upgrading to OData 8.0.1, but the breaking changes were beyond me to fix. (especially around conventions)

**Comment by dougbu** [comment]
> Looks like there are binary breaking changes in Logging: 😕

This is a known issue though the breadth of the change is pretty annoying. Fix is

``` c#
private static readonly LogDefineOptions SkipEnabledCheckLogOptions = new() { SkipEnabledCheck = true };`
...
_generatedCodeToAssemblyCompilationEnd = LoggerMessage.Define<string, double>(
    LogLevel.Annoyance,
    2,
    "...",
    SkipEnabledCheckLogOptions);
```

See also @TanayParikh's work in dotnet/aspnetcore#34033 and the actual break in dotnet/runtime#54581.

File: test/EFCore.OData.FunctionalTests/Properties/TestAssemblyConditions.cs (hunk 1 above)
Comment (smitpatel) [reply test/EFCore.OData.FunctionalTests/Properties/TestAssemblyConditions.cs:15]: Ah, missed the other comments and this is OData project.

**Comment by bricelam** [comment]
So, the problem with that approach is that we don't own the code with `skipEnabledCheck: true`. It's inside the OData package. Will skip until OData releases a new package that's compatible (or the binary breaking change is reverted).

**Comment by AndriySvyryd** [comment]
Upgrading to Preview 7 resolves this, but brings other breaking changes, see https://github.com/dotnet/efcore/pull/25324

**Comment by bricelam** [comment]
Yuck. Our branches are getting messy with one-offs in preview 6 and preview 7. ☹️ Do we have a clear path forward? Should we snub contributors and start using preview 7 (or rc 1) builds of the SDK in main?

**Comment by AndriySvyryd** [comment]
> Should we snub contributors and start using preview 7 (or rc 1) builds of the SDK in main?

We should wait until https://github.com/dotnet/runtime/issues/56285 is fixed and then jump to rc1. If it's not fixed in rc1, then we'll need to have a discussion.

**Comment by bricelam** [comment]
@AndriySvyryd It looks like the latest daily build of the Preview 7 SDK has it. Can you review to make sure I didn't keep anything from #25324 that we don't want in `main`?

**Review by AndriySvyryd — APPROVED** [review]

**Pull request merged**